js/bybit.js

```diff
@@ -3718,20 +3734,36 @@ module.exports = class bybit extends Exchange {
         } else if (timeInForce === 'ioc') {
             request['timeInForce'] = 'ImmediateOrCancel';
         }
-        const triggerPrice = this.safeValue2 (params, 'stopPrice', 'triggerPrice');
-        const stopLossPrice = this.safeValue (params, 'stopLossPrice', triggerPrice);
-        const isStopLossOrder = stopLossPrice !== undefined;
-        const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
-        const isTakeProfitOrder = takeProfitPrice !== undefined;
-        if (isStopLossOrder || isTakeProfitOrder) {
-            const triggerAt = isStopLossOrder ? stopLossPrice : takeProfitPrice;
-            const preciseTriggerPrice = this.priceToPrecision (symbol, triggerAt);
+        const triggerPrice = this.safeNumber2 (params, 'triggerPrice', 'stopPrice');
+        const stopLossTriggerPrice = this.safeNumber (params, 'stopLossPrice', triggerPrice);
+        const takeProfitTriggerPrice = this.safeNumber (params, 'takeProfitPrice');
+        const isStopLoss = this.safeValue (params, 'stopLoss');
+        const isTakeProfit = this.safeValue (params, 'takeProfit');
```

Comment:
@Dan-krm in this case, I don't think we should do this because `takeProfit` is also the exchange-specific parameter. Besides that, you can't set anything other than the price, so there is no point in encapsulating it inside an object IMO.

In other words, instead of 
`'{"stopLoss":'{"price":1200}',"takeProfit":'{"price":1800}'}'` 

I think we should stick with
`'{"stopLoss":1200,"takeProfit":1800}'`

Reply:
I agree I'll push some changes

Reply:
@Dan-krm thanks